vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -507,15 +572,35 @@ void vtkF3DRenderer::ShowGrid(bool show)
 //----------------------------------------------------------------------------
 void vtkF3DRenderer::ConfigureGridUsingCurrentActors()
 {
+  double* up = this->GetEnvironmentUp();
+  double* right = this->GetEnvironmentRight();
+  double front[3];
+  vtkMath::Cross(right, up, front);
+
+  vtkNew<vtkMatrix4x4> upMatrix;
+  {
+    const double m[16] = {
+      right[0], right[1], right[2], 0, //
+      up[0], up[1], up[2], 0,          //
+      front[0], front[1], front[2], 0, //
+      0, 0, 0, 1,                      //
+    };
+    upMatrix->DeepCopy(m);
+  }
+  vtkNew<vtkMatrix4x4> upMatrixInv;
+  upMatrixInv->DeepCopy(upMatrix);
+  upMatrixInv->Invert();
+  vtkNew<vtkTransform> upTransformInv;
+  upTransformInv->SetMatrix(upMatrixInv);
+  vtkNew<vtkTransform> upTransform;
+  upTransform->SetMatrix(upMatrix);
```

Comment:
A few remarks about that:
- You can use `vtkMatrix3x3` directly and use `MultiplyPoint` instead of relying on `vtkTransform` (which is very slow)
- The matrix is orthonormal, so you can just use the transpose instead of the inverse.